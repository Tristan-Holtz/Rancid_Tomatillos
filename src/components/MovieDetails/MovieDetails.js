import React from 'react';
import './MovieDetails.scss';

const MovieDetails = ({ location }) => {
  const movie = location.state;
  return (
    <section className="card-detail-section">
      <div className="movie-main">
        <img
          className="movie-backdrop"
          src={movie.backdrop_path}
          alt={movie.title + ' backdrop'}
        />
        <div className="overview-container">
          <label className="movie-overview">Overview</label>
          <p className="movie-description">{movie.overview}</p>
        </div>
      </div>
      <article className="movie-stats-container">
        <img
          className="movie-poster"
          src={movie.poster_path}
          alt={movie.title + ' poster'}
        />
        <div className="movie-stats">
          <h1 className="movie-title">{movie.title}</h1>
          <p className="movie-date">Release Date {movie.release_date}</p>
          <h3 className="movie-rating">
            Average Rating {movie.average_rating}
          </h3>
        </div>
      </article>
    </section>
  );
};

export default MovieDetails;
