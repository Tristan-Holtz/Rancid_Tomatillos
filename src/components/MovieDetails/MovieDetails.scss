.card-detail-section {
  display: flex;
  flex-direction: column;
  width: 100%;
}

.movie-backdrop {
  background: center center fixed;
  height: 450px;
  min-width: 900px;
  width: 900px;
}

.movie-poster {
  border: 5px solid black;
  border-radius: 8px;
  height: 350px;
  margin: -250px 0 0 25px;
  width: 225px;
}

.movie-overview {
  color: #fff;
  font-weight: bold;
}

.movie-description {
  color: #fff;
  height: 65px;
  overflow: auto;
}

.movie-stats-container {
  display: flex;
  flex-direction: row;
}

.movie-stats {
  padding-left: 2em;
}

.overview-container {
  background-color: gray;
  padding: 1em;
}

.movie-main {
  display: flex;
}
