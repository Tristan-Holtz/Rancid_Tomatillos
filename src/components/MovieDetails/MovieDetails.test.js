import React from 'react';
import { shallow } from 'enzyme';
import { MovieDetails } from './MovieDetails';

describe('MovieDetails', () => {
  it('should render the component', () => {
    const mockProps = {
      location: {
        state: {
          id: 21,
          backdrop_path: '',
          poster_path: '',
          title: 'Sanic',
          overview: 'Sanic goes fast',
          average_rating: 6,
          release_date: 'TODAY'
        }
      }
    };
    const wrapper = shallow(<MovieDetails {...mockProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
