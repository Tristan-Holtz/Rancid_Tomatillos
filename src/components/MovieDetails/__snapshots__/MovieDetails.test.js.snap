// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails should render the component 1`] = `ShallowWrapper {}`;
